File: backend/app/models/menu.py
from datetime import datetime

from sqlalchemy import Boolean, DateTime, Float, Integer, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base


class MenuItem(Base):
    __tablename__ = "menu_items"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    name: Mapped[str] = mapped_column(String(100))
    description: Mapped[str] = mapped_column(Text, nullable=True)
    price: Mapped[float] = mapped_column(Float)
    category: Mapped[str] = mapped_column(String(50), default="General")
    image_url: Mapped[str | None] = mapped_column(String(255), nullable=True)
    is_available: Mapped[bool] = mapped_column(Boolean, default=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, server_default=func.now(), onupdate=func.now()
    )

    order_items: Mapped[list["OrderItem"]] = relationship(back_populates="menu_item")

File: app/src/types/api/menu.ts
export type MenuItemApi = {
	id: number;
	name: string;
	description: string | null;
	price: number;
	category: string;
	image_url: string | null;
	is_available: boolean;
	created_at: string;
	updated_at: string;
};

export type CreateMenuItemPayload = {
	name: string;
	description?: string | null;
	price: number;
	category: string;
	is_available: boolean;
};

export type UpdateMenuItemPayload = {
	name?: string;
	description?: string | null;
	price?: number;
	category?: string;
	is_available?: boolean;
};
